infra/modules/waf/main.tf

```diff
@@ -0,0 +1,385 @@
+    override_action {
+      count {}
+    }
+    statement {
+      managed_rule_group_statement {
+        name        = "AWSManagedRulesUnixRuleSet"
+        vendor_name = "AWS"
+      }
+    }
+    visibility_config {
+      cloudwatch_metrics_enabled = true
+      metric_name                = "unix-vulnerability-metrics"
+      sampled_requests_enabled   = false
+    }
+  }
+  # the following are custom rules and arent managed by AWS.
+  rule {
+    # Applies a rate based rule to IPs originating in the US
+    name     = "AWSRateBasedRuleDomesticDOS"
+    priority = 8
+
+    action {
+      block {}
+    }
+
+    statement {
+      rate_based_statement {
+        limit              = 2000 # max number of requests per 5-minute period from a single originating IP address
+        aggregate_key_type = "IP"
+
+        scope_down_statement {
+          geo_match_statement {
+            country_codes = ["US"]
+          }
+        }
+      }
+    }
+    visibility_config {
+      cloudwatch_metrics_enabled = true
+      metric_name                = "domestic-dos-metrics"
+      sampled_requests_enabled   = false
+    }
+  }
+
+  rule {
+    # Applies a rate based rule to IPs originating outside of the US
+    name     = "AWSRateBasedRuleGlobalDOS"
+    priority = 9
+
+    action {
+      block {}
+    }
+
+    statement {
+      rate_based_statement {
+        limit              = 500 # max number of requests per 5-minute period from a single originating IP address
+        aggregate_key_type = "IP"
+
+        scope_down_statement {
+          not_statement {
+            statement {
+              geo_match_statement {
+                country_codes = ["US"]
+              }
+            }
+          }
+        }
+      }
+    }
+
+    visibility_config {
+      cloudwatch_metrics_enabled = true
+      metric_name                = "global-dos-metrics"
+      sampled_requests_enabled   = false
+    }
+  }
+  visibility_config {
+    cloudwatch_metrics_enabled = true
+    metric_name                = "waf-general-metrics"
+    sampled_requests_enabled   = false
+  }
+}
+
+# logging configuration resource
+resource "aws_wafv2_web_acl_logging_configuration" "waf_logging" {
+  log_destination_configs = [aws_kinesis_firehose_delivery_stream.waf_logging.arn]
+  resource_arn            = aws_wafv2_web_acl.waf.arn
+}
+# firehose to recieve logs
+resource "aws_kinesis_firehose_delivery_stream" "waf_logging" {
+  name        = "aws-waf-logs-metrics-stream"
+  destination = "extended_s3"
+  server_side_encryption {
+    enabled  = true
+    key_type = "CUSTOMER_MANAGED_CMK"
+    key_arn  = module.s3_encrypted_bucket.bucket_kms_arn
+    # key_arn  = aws_kms_key.waf_logging.arn
+  }
+  extended_s3_configuration {
+    role_arn   = aws_iam_role.firehose_perms.arn
+    bucket_arn = module.s3_encrypted_bucket.encrypted_bucket_arn
+  }
+}
+
+# IAM Role for Kinesis
+resource "aws_iam_role" "firehose_perms" {
+  name               = "waf-firehose-role"
```

Comment:
We should actually make this a variable so that in `env-template/main.tf` we can set a local that prepends the `"${local.project_name}"`

Reply:
That way in the [IAM console](https://us-east-1.console.aws.amazon.com/iamv2/home?region=us-west-2#/roles), we can search for "wic-prp" and see all the roles we have defined